tests/kernel/mbox/mbox_api/src/test_mbox_api.c

```diff
@@ -605,86 +604,102 @@ void test_define_multi_mbox(void)
 	tmbox(&mbox3);
 }
 
-void test_mbox_put_get_null(void)
+ZTEST(mbox_api, test_mbox_put_get_null)
 {
 	info_type = PUT_GET_NULL;
 	tmbox(&mbox);
 }
 
-void test_mbox_put_get_buffer(void)
+ZTEST(mbox_api, test_mbox_put_get_buffer)
 {
 	info_type = PUT_GET_BUFFER;
 	tmbox(&mbox);
 }
 
-void test_mbox_async_put_get_buffer(void)
+ZTEST(mbox_api, test_mbox_async_put_get_buffer)
 {
 	info_type = ASYNC_PUT_GET_BUFFER;
 	tmbox(&mbox);
 }
 
-void test_mbox_async_put_get_block(void)
+ZTEST(mbox_api, test_mbox_async_put_get_block)
 {
 	info_type = ASYNC_PUT_GET_BLOCK;
 	tmbox(&mbox);
 }
 
-void test_mbox_target_source_thread_buffer(void)
+ZTEST(mbox_api, test_mbox_target_source_thread_buffer)
 {
 	info_type = TARGET_SOURCE_THREAD_BUFFER;
 	tmbox(&mbox);
 }
 
-void test_mbox_incorrect_receiver_tid(void)
+ZTEST(mbox_api, test_mbox_incorrect_receiver_tid)
 {
 	info_type = INCORRECT_RECEIVER_TID;
 	tmbox(&mbox);
 }
 
-void test_mbox_incorrect_transmit_tid(void)
+ZTEST(mbox_api, test_mbox_incorrect_transmit_tid)
 {
 	info_type = INCORRECT_TRANSMIT_TID;
 	tmbox(&mbox);
 }
 
-void test_mbox_timed_out_mbox_get(void)
+ZTEST(mbox_api, test_mbox_timed_out_mbox_get)
 {
 	info_type = TIMED_OUT_MBOX_GET;
 	tmbox(&mbox);
 }
 
-void test_mbox_msg_tid_mismatch(void)
+ZTEST(mbox_api, test_mbox_msg_tid_mismatch)
 {
 	info_type = MSG_TID_MISMATCH;
 	tmbox(&mbox);
 }
 
-void test_mbox_dispose_size_0_msg(void)
+ZTEST(mbox_api, test_mbox_dispose_size_0_msg)
 {
 	info_type = DISPOSE_SIZE_0_MSG;
 	tmbox(&mbox);
 }
 
-void test_mbox_async_put_to_waiting_get(void)
+ZTEST(mbox_api, test_mbox_async_put_to_waiting_get)
 {
 	info_type = ASYNC_PUT_TO_WAITING_GET;
 	tmbox(&mbox);
 }
 
-void test_mbox_get_waiting_put_incorrect_tid(void)
+ZTEST(mbox_api, test_mbox_get_waiting_put_incorrect_tid)
 {
 	info_type = GET_WAITING_PUT_INCORRECT_TID;
 	tmbox(&mbox);
 }
 
-void test_mbox_async_multiple_put(void)
+ZTEST(mbox_api, test_mbox_async_multiple_put)
 {
 	info_type = ASYNC_MULTIPLE_PUT;
 	tmbox(&mbox);
 }
 
-void test_mbox_multiple_waiting_get(void)
+ZTEST(mbox_api, test_mbox_multiple_waiting_get)
 {
 	info_type = MULTIPLE_WAITING_GET;
 	tmbox(&mbox);
+
+	/* cleanup the sender threads */
+	for (int i = 0; i < 5 ; i++) {
+		k_thread_abort(&waiting_get_tid[i]);
```

Comment:
One of the interesting side effects of ztest2 re-ordering tests is that it's forced us to fix a bunch of state/cleanup issues like this one with existing tests.  I actually was thinking this was a misfeature (it makes it hard to find "the test you are working on" in output when developing), but now I'm thinking maybe we want to go further and have a "CONFIG_ZTEST_RANDOMIZE_TEST_ORDER" or something for use in CI/stress environments to further harden the tests.  Something to consider, @yperess?

Reply:
Your concern is totally right again, @andyross . I think we have "CONFIG_ZTEST_SHUFFLE" in place to make test running in a random order. And your concern is totally right , this test cannot fully passing the testing every time with this config set. I will fix it ASAP. Thanks for pointing this out!